### ISSUE TITLE
RateCard API - parameters

### ISSUE THREAD

**Issue by kubal5003** [Azure/BillingCodeSamples#18]
The parameters that are required in order to get the data from RateCardApi are undocumented and confusing.

1) Where can I get the information about OfferId for a given subscription? Why do I need to even care about this? If you wanted OfferDurableId then why am I supposed to pass SubscriptionId? You clearly have an internal API for querying that data for subscription - Azure Portal is able to present that information.

Is there any way I could query it? Or if not please remove that parameter and make it automatic.

2) Locale - please set the default to en-US and don't make it required. It's a nice to have feature, but it doesn't make sense to make it required. Also please specify a list of acceptable values unless any combination is fine (by any combination I mean: http://timtrott.co.uk/culture-codes/ ).

3) Currency - again this is a nice to have parameter, but you already have that information stored in the subscription. There is documentation about Azure supported currencies, but there's no API that would guarantee the information to be up to date. ( https://azure.microsoft.com/en-us/pricing/faq/ )

4) Region Info - To quote the docs: "Set {RegionInfo} to the 2 letter ISO code where the offer was purchased". What are the available options? If I'm from Poland should I pass PL or rather EU  since my currency is EUR ? 

To sum up - all those parameters are unnecessary & confusing. Please either:
- set the defaults based on subscription data
- expose a public API to be able to query that data given subscription id
- or at least document the available options or expose metadata for this service

I'm developing a free to use service for everyone to be able to take a look at their usage. Querying the usage API was pretty nice - the UX was login & go, but the RateCard API is unfortunately far from perfect and ruined my weekend.

**Comment by XristophD** [comment]
Hello @kubal5003 
I know, it has been a while you posted that, but I just stumbled about the same question and wanted to know if you figured out the RegionInfo issue? What are we supposed to send as filter for region? 

Could not find any documentation either, the RateCard API is in preview since over a year now.

**Comment by kubal5003** [comment]
I did not figure this out. I have been contacted by numerous Azure people and was trying to follow up with them, but I think it never got to the Billing Team to anyone who could actually answer. As you can also see here - no luck. If you're not a rich company it's impossible to get any information back from Microsoft.. Sad, but true. I stopped investigating this topic, because it's impossible to get any answers about this and also I had issues with the whole Azure AD and how to properly configure it for multi-tenant access.. Then MS announced that they bought one of the companies providing insights into Azure (Cloud Cruiser? Cloudyn? I don't remember which one..), so I just left the topic to die. If there's no answer for a fundamental question on how to use the API then they can as well hide this API in their private network and don't even bother with github.

**Comment by renutz** [comment]
+1. We also need a good RateCard API working with PAYG/MPN, CSP and EA. There are too many sources, too many ways and too many problems.
